app/assets/javascripts/admin/addon/controllers/admin-badges/show.js

```diff
@@ -0,0 +1,238 @@
+import Controller, { inject as controller } from "@ember/controller";
+import { observes } from "discourse-common/utils/decorators";
+import I18n from "I18n";
+import bootbox from "bootbox";
+import { bufferedProperty } from "discourse/mixins/buffered-content";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { next } from "@ember/runloop";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import getURL from "discourse-common/lib/get-url";
+import { tracked } from "@glimmer/tracking";
+
+const IMAGE = "image";
+const ICON = "icon";
+
+// TODO: Stop using Mixin here
+export default class AdminBadgesShowController extends Controller.extend(
+  bufferedProperty("model")
+) {
+  @controller adminBadges;
+  @service router;
+
+  @tracked saving = false;
+  @tracked savingStatus = "";
```

Comment:
Interesting, looks like using decorators and fields initialized with strings breaks syntax highlighting on github? 👀  (following ~70 lines are borked)

Reply:
Strange!